packages/config-storybook/src/main.ts

```diff
@@ -0,0 +1,103 @@
+import ResolveTypescriptPlugin from 'resolve-typescript-plugin'
+import type { Configuration as WebpackConfig, RuleSetRule } from 'webpack'
+
+export interface EssexStorybookConfig {
+	stories?: string[]
+	staticDirs?: string[]
+	resolveAliases?: Record<string,string>
+	transpileMatch: (string | RegExp)[]
+}
+
+const DEFAULT_STORIES = [
+	/**
+	 * search sibling packages by default 
+	 */
+	'../../*/src/**/*.stories.@(mdx|js|jsx|ts|tsx)'
+]
+const DEFAULT_STATIC_DIRS: string[] = []
+
+export function configure({
+	stories = DEFAULT_STORIES,
+	staticDirs = DEFAULT_STATIC_DIRS,
+	resolveAliases = {},
+	transpileMatch = [],
+}: EssexStorybookConfig) {
+	return {
+		stories,
+		staticDirs,
+		addons: [
+			'@storybook/addon-links',
+			'@storybook/addon-essentials',
+			'@storybook/addon-interactions',
+		],
+		framework: '@storybook/react',
+		typescript: {
+			reactDocgen: 'react-docgen-typescript',
+			reactDocgenTypescriptOptions: {
+				compilerOptions: {
+					allowSyntheticDefaultImports: false,
+					esModuleInterop: false,
+				},
+			},
+		},
+		webpackFinal(config: WebpackConfig) {
+			// mute build output
+			if (process.env['CI'] || process.env['SB_QUIET']) {
+				config.stats = 'errors-only'
+				config.plugins = config.plugins?.filter(
+					({ constructor }: any) => constructor.name !== 'ProgressPlugin',
+				)
+			}
+
+			if (!config.resolve) {
+				config.resolve = {}
+			}
+
+			config.resolve.alias = {
+				...(config.resolve.alias ?? {}),
+				...resolveAliases,
+				'styled-components': require.resolve('styled-components'),
```

Comment:
Has whatever issue styled-components had that led us to create @essex/styled-components been resolved in newer versions?

Reply:
Not sure, but we use it everywhere in our component libraries, and I think we should just go with it for now.